DEVELOPER.md

```diff
@@ -0,0 +1,114 @@
+# Notes for Development
+
+## Development Environment
+
+## Building from Source
+
+To build and test the driver, the standard cargo commands can be used from the root directory.
+
+For an unoptimized build with debugging information (most common):
+```
+cargo build
+```
+
+For an optimized build with debugging info (mac)
+```
+cargo build --features odbc-sys/iodbc,cstr/utf32 --profile=release-with-debug
+```
+And for an optimized build with debugging info (windows, linux)
+```
+cargo build --profile=release-with-debug
+```
+The resulting build files will be output to the `target` directory in the `debug` or `release` folders respectively.
+
+## Setting up the driver manager on MacOS
+
+For macos, we use [iodbc](https://www.iodbc.org/dataspace/doc/iodbc/wiki/iodbcWiki/WelcomeVisitors) as the driver manager. To set up iODBC, first download and install it
+```
+export $INSTALLED_ODBC_PATH=$PWD/installed_odbc/install"
+mkdir -p "$INSTALLED_ODBC_PATH"
+cd installed_odbc
+echo "downloading iODBC"
+iODBC_dir=libiodbc-3.52.15
+curl -LO "https://github.com/openlink/iODBC/releases/download/v3.52.15/$iODBC_dir.tar.gz" \
+--silent \
+--fail \
+--max-time 60 \
+--retry 5 \
+--retry-delay 0
+
+tar xf "$iODBC_dir.tar.gz"
+cd "$iODBC_dir"
+./configure --prefix="$INSTALLED_ODBC_PATH"
+make 
+make install
+```
+Next, using the steps above, compile the release version with debug info. Additionally, set the library paths:
+```
+export LIBRARY_PATH="$INSTALLED_ODBC_PATH/lib"
+export LD_LIBRARY_PATH="$INSTALLED_ODBC_PATH/lib"
+export DYLD_LIBRARY_PATH="$INSTALLED_ODBC_PATH/lib"
```

Comment:
also wanted to note that our conclusion from earlier was correct, and adding this one line to my personal setup (after following the rest of these steps) fixed the problem.